src/core/document/json.pl

```diff
@@ -1885,17 +1900,21 @@
 type_id_predicate_iri_value(optional(C),Id,P,O,Recursion,DB,Prefixes,Compress_Ids,Unfold,V) :-
     type_descriptor(DB,C,Desc),
     type_id_predicate_iri_value(Desc,Id,P,O,Recursion,DB,Prefixes,Compress_Ids,Unfold,V).
-type_id_predicate_iri_value(base_class(C),_,_,X^^T,_,_,Prefixes,_Compress,_Unfold,V) :-
+type_id_predicate_iri_value(base_class(C),_,_,Elt,_,DB,Prefixes,_Compress,_Unfold,V) :-
     % NOTE: This has to treat each variety of JSON value as natively
     % as possible.
-    (   C = T % The type is not just subsumed but identical - no ambiguity.
-    ->  value_type_json_type(X,T,V,_)
-    ;   value_type_json_type(X,T,D,T2),
-        % NOTE: We're always compressing, even if Compress_Ids is false
-        % The reason here is that this is a datatype property, not a node of our own.
-        % We may want to revisit this logic though.
-        compress_dict_uri(T2,Prefixes,T2C),
-        V = json{ '@type' : T2C, '@value' : D}
+    (   C = 'http://terminusdb.com/schema/sys#JSON'
+    ->  get_json_object(DB, Elt, V)
+    ;   Elt = X^^T,
+        (   C = T % The type is not just subsumed but identical - no ambiguity.
+        ->  value_type_json_type(X,T,V,_)
+        ;   value_type_json_type(X,T,D,T2),
+            % NOTE: We're always compressing, even if Compress_Ids is false
+            % The reason here is that this is a datatype property, not a node of our own.
+            % We may want to revisit this logic though.
+            compress_dict_uri(T2,Prefixes,T2C),
+            V = json{ '@type' : T2C, '@value' : D}
+        )
```

Comment:
could be more defensive to throw here if `C` is not a json and also `Elt` is not a `X^^T`.